src/libcoretest/cell.rs

```diff
@@ -159,3 +159,27 @@ fn refcell_default() {
     let cell: RefCell<u64> = Default::default();
     assert_eq!(0, *cell.borrow());
 }
+
+#[test]
+fn unsafe_cell_unsized() {
+    let cell: &UnsafeCell<[i32]> = &UnsafeCell::new([1, 2, 3]);
+    {
+        let val: &mut [i32] = unsafe { &mut *cell.get() };
+        val[0] = 4;
+        val[2] = 5;
+    }
+    let comp: &mut [i32] = &mut [4, 5, 3];
+    assert_eq!(unsafe { &mut *cell.get() }, comp);
```

Comment:
So there was just a typo here causing the failure, but I'm thinking you have aliasing `&mut` pointers here, that's generally very invalid and UB, is it possible to write these comparisons with shared pointers?